src/scenic/core/object_types.py

```diff
@@ -3,666 +3,1760 @@
-	a standard deviation given by the ``positionStdDev`` property.
-
-	Properties:
-		position (`Vector`; dynamic): Position of the point. Default value is the origin.
-		visibleDistance (float): Distance for :keyword:`can see` operator. Default value 50.
-		width (float): Default value zero (only provided for compatibility with
-		  operators that expect an `Object`).
-		length (float): Default value zero.
-		mutationScale (float): Overall scale of mutations, as set by the
-		  :keyword:`mutate` statement. Default value zero (mutations disabled).
-		positionStdDev (float): Standard deviation of Gaussian noise to add to this
-		  object's :prop:`position` when mutation is enabled with scale 1. Default value 1.
-	"""
-	position: PropertyDefault((), {'dynamic'}, lambda self: Vector(0, 0))
-	width: 0
-	length: 0
-	visibleDistance: 50
-
-	mutationScale: 0
-	mutator: PropertyDefault({'positionStdDev'}, {'additive'},
-							 lambda self: PositionMutator(self.positionStdDev))
-	positionStdDev: 1
-
-	# This property is defined in Object, but we provide a default empty value
-	# for Points for implementation convenience.
-	regionContainedIn: None
-
-	@cached_property
-	def visibleRegion(self):
-		"""The :term:`visible region` of this object.
-
-		The visible region of a `Point` is a disc centered at its ``position`` with
-		radius ``visibleDistance``.
-		"""
-		return CircularRegion(self.position, self.visibleDistance)
-
-	@cached_property
-	def corners(self):
-		return (self.position,)
-
-	def toVector(self) -> Vector:
-		return self.position
-
-	def canSee(self, other) -> bool:	# TODO improve approximation?
-		for corner in other.corners:
-			if self.visibleRegion.containsPoint(corner):
-				return True
-		return False
-
-	def sampleGiven(self, value):
-		sample = super().sampleGiven(value)
-		if self.mutationScale != 0:
-			for mutator in self.mutator:
-				if mutator is None:
-					continue
-				sample, proceed = mutator.appliedTo(sample)
-				if not proceed:
-					break
-		return sample
-
-	# Points automatically convert to Vectors when needed
-	def __getattr__(self, attr):
-		if hasattr(Vector, attr):
-			return getattr(self.toVector(), attr)
-		else:
-			raise AttributeError(f"'{type(self).__name__}' object has no attribute '{attr}'")
+    """The Scenic base class ``Point``.
+
+    The default mutator for `Point` adds Gaussian noise to ``position`` with
+    a standard deviation given by the ``positionStdDev`` property.
+
+    Properties:
+        position (`Vector`; dynamic): Position of the point. Default value is the origin (0,0,0).
+        width (float): Default value 0 (only provided for compatibility with
+          operators that expect an `Object`).
+        length (float): Default value 0.
+        height (float): Default value 0.
+        baseOffset (`Vector`): An offset indicating where the base of this `Point` is relative to
+          its position. Used by the ``on`` specifier. Default value is (0,0,0).
```

Comment:
This seems like another property that really should be added by `Object` but which is put on `Point` to simplify the implementation (since how can a point have a "base"?). If so then it would be better to use a very short description like that for `width` and refer to `Object` for the real definition.